Moved PropertyContainerCreateFromLua() to dmsdk

Gert-Jan opened the pull request:
Added `dmScript:: PropertyContainerCreateFromLua()` to `dmsdk/gameobject/script.h`

### Technical changes
* Placed `PropertyContainerCreateFromLua()` in `dmsdk/gameobject/script.h` [per request](https://github.com/defold/defold/pull/10800#discussion_r2137476565), left the implementation in the dmGameObject namespace, because it contains a great number of references within the dmGameObject namespace.

Gert-Jan committed:
Moved PropertyContainerCreateFromLua to dmsdk.

Gert-Jan committed:
Merge branch 'defold:dev' into feature/dmsdk-property-container-from-lua

Gert-Jan committed:
Fixed comments.

Gert-Jan committed:
Added documentation.

JCash commented on engine/gameobject/src/dmsdk/gameobject/gameobject_props_lua.h:
```diff
@@ -0,0 +1,33 @@
+typedef struct lua_State lua_State;
+namespace dmGameObject
+{
+    /*#
+     * Creates a property container from a lua table
+     * @name PropertyContainerCreateFromLua
+     * @param L [type: lua_State*] The lua state
+     * @param index [type: int] The lua stack index of the lua table
+     * @return container [type: HPropertyContainer] The property container
+     */
+    HPropertyContainer PropertyContainerCreateFromLua(lua_State* L, int index);
```
As this is a new file, it would need a @document paragraph (see e.g. gameobject.h)

However, I think we should move this function to gameobject/src/dmsdk/gameobject/script.h as it's part of the scripting api.
That would of course make it `dmScript:: PropertyContainerCreateFromLua()`, but I think that's a good change.

Gert-Jan replied:
Done and updated the PR description.

JCash reviewed (changes requested):
overall good, but I'd like the public function to be placed in the script.h file instead.

Gert-Jan committed:
Moved PropertyContainerCreateFromLua() to dmsdk/gameobject/script.h.

Gert-Jan committed:
Moved a version of PropertyContainerCreateFromLua to dmScript.

To copy the implementation of PropertyContainerCreateFromLua to dmScript would not be a great idea I think, given how dependent the implementation is on dmGameObject.

JCash reviewed (approved):
lgtm
